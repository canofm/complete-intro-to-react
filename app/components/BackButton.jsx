// @flow

import React from 'react';
import { Link } from 'react-router-dom';

const BackButton = () => (
  <h2>
    <Link to="/search">Back</Link>
  </h2>
);

export default BackButton;
