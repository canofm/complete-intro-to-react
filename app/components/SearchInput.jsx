// @flow

import React, { Component } from 'react';
import { connect } from 'react-redux';
import { setSearchTerm } from '../actions';

const mapStateToProps = state => ({ searchTerm: state.searchTerm });
const mapDispatchToProps = (dispatch: Function) => ({
  handleSearchTermChange(event) {
    dispatch(setSearchTerm(event.target.value));
  }
});

@connect(mapStateToProps,mapDispatchToProps)
class SearchInput extends Component {
  props: {
    searchTerm: string,
    handleSearchTermChange: Function
  };

  render() {
    return (
      <input
        onChange={this.props.handleSearchTermChange}
        value={this.props.searchTerm}
        type="text"
        placeholder="Search"
      />
    );
  }
}

export default SearchInput;
