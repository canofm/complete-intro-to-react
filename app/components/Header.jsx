// @flow

import React from 'react';
import { Link } from 'react-router-dom';
import BackButton from './BackButton';
import SearchInput from './SearchInput';

const Header = (props: { showSearch?: boolean }) => {
  const utilSpace = props.showSearch ? <SearchInput /> : <BackButton />;

  return (
    <header>
      <h1>
        <Link to="/">svideo</Link>
      </h1>
      {utilSpace}
    </header>
  );
};

Header.defaultProps = {
  showSearch: false
};

export default Header;
