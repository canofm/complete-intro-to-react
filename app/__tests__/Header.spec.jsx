import React from 'react';
import { shallow } from 'enzyme';
import BackButton from '../components/BackButton';
import Header from '../components/Header';
import SearchInput from '../components/SearchInput';

describe('Header component', () => {
  describe('by default', () => {
    const component = shallow(<Header />);

    it('should render correctly', () => {
      expect(component).toMatchSnapshot();
    });
    it('should render BackButton by default', () => {
      component.contains(<BackButton />);
    });
  });
  describe('when it receive showSearch prop', () => {
    const component = shallow(<Header showSearch />);
    it('should render correctly', () => {
      expect(component).toMatchSnapshot();
    });
    it('should render SearchInput if it gets showSearch prop', () => {
      component.contains(<SearchInput />);
    });
  });
});
