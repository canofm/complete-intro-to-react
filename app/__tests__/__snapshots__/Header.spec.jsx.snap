// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component by default should render correctly 1`] = `
<header>
  <h1>
    <Link
      replace={false}
      to="/"
    >
      svideo
    </Link>
  </h1>
  <BackButton />
</header>
`;

exports[`Header component when it receive showSearch prop should render correctly 1`] = `
<header>
  <h1>
    <Link
      replace={false}
      to="/"
    >
      svideo
    </Link>
  </h1>
  <SearchInput
    handleSearchTermChange={[Function]}
    searchTerm=""
    showSearch={true}
  />
</header>
`;
